sdk/core/azure-core/inc/context.hpp

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <assert.h>
+#include <chrono>
+#include <memory>
+#include <stdio.h>
+#include <stdlib.h>
```

Comment:
`assert.h`, `stdio.h` and `stdlib.h` are C headers, we should use `cassert`, `cstdio` and `cstdlib` instead.

In addition, I don't think use assert in product code is a good idea.